addons/stock/models/stock_rule.py

```diff
@@ -539,11 +540,15 @@ def run_scheduler(self, use_new_cursor=False, company_id=False):
         and the availability of moves. This function is intended to be run for all the companies at the same time, so
         we run functions as SUPERUSER to avoid intercompanies and access rights issues. """
         try:
-            if use_new_cursor:
-                cr = registry(self._cr.dbname).cursor()
-                self = self.with_env(self.env(cr=cr))  # TDE FIXME
-
-            self._run_scheduler_tasks(use_new_cursor=use_new_cursor, company_id=company_id)
+            try:
+                if use_new_cursor:
+                    cr = registry(self._cr.dbname).cursor()
+                    self = self.with_env(self.env(cr=cr))  # TDE FIXME
+
+                self._run_scheduler_tasks(use_new_cursor=use_new_cursor, company_id=company_id)
+            except Exception:
+                _logger.error('Error during stock scheduler :\n%s' % traceback.format_exc())
```

Comment:
There is a flag in loggers to do that without having to import `traceback`:
```suggestion
                _logger.error("Error during stock scheduler", exc_info=True)
```